pwiz_tools/Skyline/Model/Results/SpectrumFilterPair.cs

```diff
@@ -265,40 +265,23 @@ public ExtractedSpectrum FilterQ3SpectrumList(MsDataSpectrum[] spectra, bool use
                     }
 
                     // Add the intensity values of all peaks that pass the filter
-                    double totalIntensity = extractedIntensities[targetIndex]; // Start with the value from the previous spectrum, if any
-                    double meanError =  highAcc ? meanErrors[targetIndex] : 0;
-                    for (int iNext = iPeak; iNext < mzArray.Length && mzArray[iNext] < endFilter; iNext++)
+                    var accumulator = new IntensityAccumulator(highAcc, Extractor, targetMz)

```

Comment:
This is so performance-critical that I am going to need to see a perf test on a large file the proves extraction using this new mechanism is not slower. It shouldn't be, but I hate to admit the number of times "shouldn't be" and "actually not" have turned out to differ.

Reply:
I ran DriftTimePredictionTutorial test on my machine on 2_21 and on this branch. The elapsed time for import results operation was almost identical: 3:08 and 3:10 min. Is this large enough? I can run it in a loop for a while to gather some statistic and tighten the confidence interval.

Reply:
MatchMz logic is fixed as well.

Reply:
Try the DiaSwathFull tutorial tests. The drift time tests will be dominated by spectrum loading. What you need is something where a lot of extraction gets done on each spectrum, i.e. extraction to spectrum point retrieval is high. That is not the case for the tutorial you chose. If the DiaSwathFull tests look good, I may still want one more test withsomething like the PanHuman library, i.e. something with just a lot, a lot of targets.

Reply:
This is SkylineTester output from TestDiaTtofFullSearchTutorial: `
Skyline 2.21
[14:42]   1.0   TestDiaTtofFullSearchTutorial                  (en)   0 failures, 7.46/154.8 MB, 90/578 handles, 416 sec.
[14:49]   2.0   TestDiaTtofFullSearchTutorial                  (en)   0 failures, 7.86/160.8 MB, 92/581 handles, 415 sec.
[14:56]   3.0   TestDiaTtofFullSearchTutorial                  (en)   0 failures, 7.86/164.1 MB, 91/578 handles, 413 sec.
[15:03]   4.0   TestDiaTtofFullSearchTutorial                  (en)   0 failures, 7.86/165.6 MB, 91/568 handles, 413 sec.
[15:10]   5.0   TestDiaTtofFullSearchTutorial                  (en)   0 failures, 7.86/164.5 MB, 91/563 handles, 409 sec.
[15:17]   6.0   TestDiaTtofFullSearchTutorial                  (en)   0 failures, 7.86/167.4 MB, 92/592 handles, 412 sec.
[15:24]   7.0   TestDiaTtofFullSearchTutorial                  (en)   0 failures, 7.86/171.5 MB, 92/578 handles, 431 sec.
**Mean=415.6 std=7.2**

PR branch
[13:51]   1.0   TestDiaTtofFullSearchTutorial                  (en)   0 failures, 7.46/153.6 MB, 89/576 handles, 419 sec.
[13:58]   2.0   TestDiaTtofFullSearchTutorial                  (en)   0 failures, 7.85/151.9 MB, 91/581 handles, 419 sec.
[14:05]   3.0   TestDiaTtofFullSearchTutorial                  (en)   0 failures, 7.85/156.3 MB, 91/571 handles, 423 sec.
[14:12]   4.0   TestDiaTtofFullSearchTutorial                  (en)   0 failures, 7.86/165.2 MB, 91/586 handles, 420 sec.
[14:19]   5.0   TestDiaTtofFullSearchTutorial                  (en)   0 failures, 7.85/165.0 MB, 90/562 handles, 424 sec.
[14:26]   6.0   TestDiaTtofFullSearchTutorial                  (en)   0 failures, 7.85/165.8 MB, 90/572 handles, 423 sec.
[14:33]   7.0   TestDiaTtofFullSearchTutorial                  (en)   0 failures, 7.85/157.8 MB, 91/576 handles, 419 sec.
**Mean=421.0 std=2.2**
`
Looks pretty similar to me. Where can I find those PanHuman library files?

Reply:
I recorded import times in the TestDiaTtofFullSearchTutorial as Brendan requested for 2.21 (288 +-5 sec) and this branch (286+-3 sec).  No significant difference.